src/deroff.rs

```diff
@@ -917,6 +1142,203 @@ fn test_is_white() {
 }
 
 #[test]
+fn test_var() {
+    let mut d = Deroffer::new();
+
+    // "\n" successes
+    d.s = "\\n dyHello".to_owned();
+    assert!(d.var() == true);
+    assert!(d.s == "Hello");
+
+    d.s = "\\n(aaHello".to_owned();
+    assert!(d.var() == true);
+    assert!(d.s == "Hello");
+
+    d.s = "\\n[skipme] Hello".to_owned();
+    assert!(d.var() == true);
+    assert!(d.s == "] Hello");
+
+    d.s = "\\naHello".to_owned();
+    assert!(d.var() == true);
+    assert!(d.s == "Hello");
+
+    // "\n" errors
+    d.s = "\\n".to_owned();
+    assert!(d.var() == false);
+    assert!(d.s == "\\n");
+
+    d.s = "\\n a".to_owned();
+    assert!(d.var() == false);
+    assert!(d.s == "\\n a");
+
+    d.s = "\\n da".to_owned();
+    assert!(d.var() == false);
+    assert!(d.s == "\\n da");
+
+    // "\*" successes
+
+    // these blocks are more for me to understand the code,
+    // but, I think they're probably helpful for you guys,
+    // so here they are. On another note, I cannot WAIT to
+    // reformat this entire project
+
+    /*
+    AA is two non whitespace characters
+    "\*(AA" => {
+        if AA in self.reg_table => {
+            self.s = self.reg_table.get(AA);
+            return true;
+        } else => {
+            self.s = self.s.get(5..);
+            return false;
+        }
+    } */
+    d.s = "\\*(traaaaaaaaaaaaa".to_owned();
+    d.reg_table
+        .insert("tr".to_owned(), "Hello World!".to_owned());
+    assert!(d.var() == true);
+    assert!(d.s == " World!");
+    let o = d.output.take();
+    assert!(o.contains("Hello"));
+    d.output.set(o);
+
+    d.s = "\\*(aaHello World!".to_owned();
+    assert!(d.var() == false);
+    assert!(d.s == "Hello World!");
+
+    /*
+    A is a string that does not start with whitespace
+    "\*[A" => {
+        if A ends with "]" => {
+            let B = A[..-1];
+
+            if B in self.reg_table {
+                self.s = self.reg_table.get(B);
+                return true;
+            } else {
+                +1 to skip the "]" as well
+                self.s = self.s[len(B)+1..];
+                return false;
+            }
+        } else => {
+            self.s = "";
+            return false;
+        }
+    }
+    */
+
+    // ideal case, B is in reg_table
+    d.s = "\\*[test_reg]".to_owned();
+    d.reg_table
+        .insert("test_reg".to_owned(), "It me!".to_owned());
+    assert!(d.var() == true);
+    assert!(d.s == " me!");
+    let o = d.output.take();
+    assert!(o.contains("It"));
+    d.output.set(o);
+
+    // no "]"
+    d.s = "\\*[foo bar :)".to_owned();
+    assert!(d.var() == false);
+    assert!(d.s == "");
+
+    // B not in reg_table
+    d.s = "\\*[foo bar]abcd".to_owned();
+    assert!(d.var() == false);
+    assert!(d.s == "abcd");
+
+    // Here's a python version of these tests
```

Comment:
Can remove all this commented-out code. Maybe something to keep on a separate branch?

Reply:
Removed